packages/cli/src/lib/packager/productionPack.ts

```diff
@@ -147,20 +151,39 @@ async function prepareExportsEntryPoints(
   >();
 
   const entryPoints = readEntryPoints(pkg);
+  const project = await createTypeDistProject();
+
   for (const entryPoint of entryPoints) {
     if (!SCRIPT_EXTS.includes(entryPoint.ext)) {
       outputExports[entryPoint.mount] = entryPoint.path;
       continue;
     }
+
     const exp = {} as Record<string, string>;
     for (const [key, ext] of Object.entries(EXPORT_MAP)) {
       const name = `${entryPoint.name}${ext}`;
       if (distFiles.includes(name)) {
         exp[key] = `./${posixPath.join(`dist`, name)}`;
       }
     }
+
     exp.default = exp.require ?? exp.import;
 
+    if (exp.types) {
+      const defaultFeatureType =
+        pkg.backstage?.role &&
+        getEntryPointDefaultFeatureType(
+          pkg.backstage?.role,
+          packageDir,
+          project,
+          exp.types,
+        );
+
+      if (defaultFeatureType) {
+        exp.backstage = defaultFeatureType;
```

Comment:
Could you refactor to move this to the top? something like `exp = { backstage: defaultFeatureType, ...exp }`. I'm a bit worried some tools might treat this differently by being last.

Reply:
Mmm I don't think it can. It needs to come after when the `exp.types` has been added as that path is used to search for the default export type.

Reply:
This section: https://nodejs.org/docs/latest-v20.x/api/packages.html#conditional-exports

In particular this bit:

> Within the "exports" object, key order is significant. During condition matching, earlier entries have higher priority and take precedence over later entries. The general rule is that conditions should be from most specific to least specific in object order.

My interpretation of that is that we want to list the `backstage` field early, because it's a very specific extension. Happy to discuss though :sweat_smile:

Reply:
Ohhh I see, just the property at the top. I thought you mean you wanted the code moved up. Sure no problem.